rhv_checks.py

```diff
@@ -338,6 +340,54 @@ def check_vms_distributed_hosts(system, warn=5, crit=10):
         sys.exit(3)
 
 
+def is_service_active(ssh, name):
+    """helper function for check_services_running"""
+    stdin, stdout, stderr = ssh.exec_command("systemctl status {}".format(name))
+    status = stdout.read()
+    return "Active: active (running)" in status
+
+
+def check_services_running(system):
+    """Check to see if the vdsm, ovirt-ha-agent, ovirt-ha-broker service is up/running"""
+    hosts = system.api.system_service().hosts_service()
+    hosts_agents = dict()
+    hosts_status = dict()
+    for host in hosts.list():
+        host_service = hosts.host_service(host.id)
+        nics = host_service.nics_service().list()
+        ip_addresses = [n.ip.address for n in nics if n.ip.address.startswith('10.')]
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        for ip in ip_addresses:
+            try:
+                ssh.connect(ip, username="root", password=system.api._password)
+                break
+            except:
+                continue
+        vdsmd = is_service_active(ssh, "vdsmd")
+        ovirt_ha_agent = is_service_active(ssh, "ovirt-ha-agent")
+        ovirt_ha_broker = is_service_active(ssh, "ovirt-ha-broker")
+        ssh.close()
+
+        hosts_agents[host.name] = {"vdsmd": vdsmd,
+                                   "ovirt-ha-agent": ovirt_ha_agent,
+                                   "ovirt-ha-broker": ovirt_ha_broker}
+        hosts_status[host.name] = all([vdsmd, ovirt_ha_agent, ovirt_ha_broker])
+
+    overall_status = all(status is True for status in hosts_status.values())
+
+    if overall_status:  # all true, everything is running
+        msg = "Ok: vdsm, ovirt-ha-agent, ovirt-ha-broker service are up/running on all hosts"
+        logger.info(msg)
+        print(msg)
+        sys.exit(0)
+    else:
+        msg = "Critical: {}".format(hosts_agents)
```

Comment:
Thoughts on printing out the host with problems? And then printing out the status of all hosts.

Reply:
Maybe we can also log the relevant messages.